### PR TITLE
Show time tooltip for DVR streams

### PR DESCRIPTION
**Pull request by donato**
This involves calculating and formatting the time offset based on the
negative duration given.

JW7-1368

### CODE REVIEW

File: src/js/view/components/timeslider.js [hunk 1]
```diff
@@ -157,11 +157,22 @@ define([
...
             var timetipText;
             if (this.activeCue) {
                 timetipText = this.activeCue.text;
             } else {
-                timetipText = utils.timeFormat(time);
+                var allowNegativeTime = true;
+                timetipText = utils.timeFormat(time, allowNegativeTime);
+
+                // If DVR and within one second of live
+                if (duration < 0 && (Math.abs(time) < 1) ) {
```
Comment (robwalch) [inline src/js/view/components/timeslider.js]: Does this work with DVR stream time events?

"within one second of live" `if (duration < 0 && (Math.abs(time) < 1)` 

I would expect it to be more like within 25-30 seconds is "Live" and 1 second of latency is very rare or unlikely.

Comment (donato) [inline src/js/view/components/timeslider.js]: As discussed in person, this is just for the text inside of the tooltip.

File: src/js/utils/parser.js [hunk 2]
```diff
@@ -116,16 +116,20 @@ define([
...
+    parser.timeFormat = function(sec, allowNegative) {
+        if (sec <= 0 && !allowNegative) {
             return '00:00';
         }
+
+        // If negative add a minus sign
+        var prefix = (sec < 0) ? '-' : '';
+        sec = Math.abs(sec);
+
+        var hrs  = Math.floor(sec / 3600),
```
Comment (donato) [inline src/js/utils/parser.js]: *technically* we should be doing Math.ceil if the time is negative

**Comment by donato** [comment]
Waiting to merge for @pajong 's unit test change

**Comment by donato** [comment]
test this please

**Pull request merged**